crates/bevy_pbr/src/render/pbr_lighting.wgsl

```diff
@@ -625,5 +746,31 @@ fn directional_light(
     color = (diffuse + specular_light) * derived_input.NdotL;
 #endif  // STANDARD_MATERIAL_CLEARCOAT
 
-    return color * (*light).color.rgb;
+    var cookie_sample = 1f;
+
+#ifdef CLUSTERED_DECALS_ARE_USABLE
+    if (*light).decal_index != 0xFFFFFFFFu {
+        let local_position = (view_bindings::clustered_decals.decals[(*light).decal_index].local_from_world *
+            vec4((*input).P, 1.0)).xyz;
+        let decal_uv = local_position.xy * vec2(-0.5, 0.5) + 0.5;
+
+        // if tiled or within tile
+        if (view_bindings::clustered_decals.decals[(*light).decal_index].tag != 0u)
+                || all(clamp(decal_uv, vec2(0.0), vec2(1.0)) == decal_uv)
+        {
+            let image_index = view_bindings::clustered_decals.decals[(*light).decal_index].image_index;
+
+            cookie_sample = textureSampleLevel(
+                view_bindings::clustered_decal_textures[image_index],
+                view_bindings::clustered_decal_sampler,
+                decal_uv - floor(decal_uv),
+                0.0
+            ).r;                    
+        } else {
+            cookie_sample = 0f;
+        }
+    }
+#endif
+
+    return color * (*light).color.rgb * cookie_sample;
 }
```

Comment:
Let's put light cookie support behind an ifdef which is only activated when light cookies are present. As it is, this is probably going to bloat our VGPR usage considerably, which is much too high already.

Reply:
i added a feature flag (`pbr_light_textures`) as doing it any other way was difficult. i also added `pbr_clustered_decals` at the same time (i guess vgpr is similar).